### ISSUE TITLE
Extract overlay displays values from external files and explodes when those are moved

### ISSUE THREAD

**Issue by ddenisenko** [mulesoft/api-workbench#60]
Steps:
1) Create RAML file with the following contents:
```RAML
#%RAML 1.0
title: Blah
version: v1
baseUri: https://api.instagram.com/{version}/
mediaType: application/json

uses:
  - testLibrary: !include testLibrary.raml

annotationTypes:
  assertion:
    parameters:
      value:


securitySchemes:
  - oauth_2_0: !include ../securitySchemes/oauth_2_0.raml
  - clientId: !include ../securitySchemes/clientId.raml
schemas:
  - location: !include ../schemas/location-schema.json
  - locations: !include ../schemas/locations-schema.json
resourceTypes:
  - base: !include ../resourceTypes/base.raml
  - mediaSearch: !include ../resourceTypes/mediaSearch.raml
traits:
  - acceptCallback: !include ../traits/acceptCallback.raml
  - limitableByGeoPoint: !include ../traits/limitableByGeoPoint.raml
types:
  TestType:
      properties:
        name: string
        name2: string
        name3: string
  TestType2:
      type: TestType
      properties:
        name4: string
  TestType3:
      type: TestType2
      properties:
        name5: string
/helloWorldddd1234ddd:
  type: base
  get:
    responses:
      200:
        body:
          application/json:
            type: TestType

```

2) Create traits folder and traits/limitableByGeoPoint.raml file with the following contents:
```RAML
queryParameters: 
  lat: 
    description: Latitude of the center search coordinate. If used, lng is required.
    type: number
  lng: 
    description: Longitude of the center search coordinate. If used, lat is required.
    type: number
  distance: 
    description: Default is 1000m (distance=1000), max distance is 5000.
    type: integer
    maximum: 5000
    default: 1000

```

3) Activate `Extract Overlay` action and choose `limitableByGeoPoint/lat` component in the left panel, move it to the right panel.

Results: exception:
```
TypeError: node.definition(...).name is not a function
    at ExtractOverlayDialog.AbstractlMoveElementsDialog.getNodeId (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:281:75)
    at ExtractOverlayDialog.AbstractlMoveElementsDialog.getNodeId (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:271:33)
    at ExtractOverlayDialog.AbstractlMoveElementsDialog.nodesEqualById (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:289:21)
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:307:34
    at Function.findIndex (/Users/munch/work/apiworkbench/api-workbench/node_modules/underscore/underscore.js:620:13)
    at Function._.find._.detect (/Users/munch/work/apiworkbench/api-workbench/node_modules/underscore/underscore.js:214:15)
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:306:46
    at Array.forEach (native)
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:305:21
    at Array.forEach (native)
    at ExtractOverlayDialog.AbstractlMoveElementsDialog.createMovedNodesTree (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:294:25)
    at ExtractOverlayDialog.AbstractlMoveElementsDialog.moveRight (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:483:18)
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/extractElementsDialog.js:366:19
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:404:68
    at Array.forEach (native)
    at HTMLButtonElement.BasicComponent.customize.element.onclick (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:404:31)
```

Setting to high as this is probably a general problem.

**Comment by ddenisenko** [comment]
Does not explode any more after my include saving fix, and allows adding overlay for an included contents.

Not that we really need this feature, but let it leave for now and see if it is going to cause any troubles.

**Comment by ddenisenko** [comment]
Fix should be available in 0.8.8.
